src/mmrelay/migrate.py

```
@@ -0,0 +1,1883 @@
+
+        report["success"] = True
+    except (MigrationError, OSError, IOError, sqlite3.DatabaseError) as exc:
+        report["success"] = False
+        report["error"] = str(exc)
+        report["message"] = "Migration failed"
+
+        # Log detailed failure info
+        staging_dir = new_home / STAGING_DIRNAME
+        logger.error(
+            "Migration failed during step: %s", getattr(exc, "step", "unknown")
+        )
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)